tests/toolkit/res_buf_tests.cc

```diff
@@ -355,5 +356,46 @@ TEST_F(ResBufTest, PushAll_DuplicateEmpty) {
   EXPECT_DOUBLE_EQ(store_.quantity(), mat1_->quantity() + mat2_->quantity());
 }
 
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+// Special tests for material buffers
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+TEST_F(MaterialBufTest, NonBulkTest) {
+
+  ASSERT_EQ(mat_store_.count(), 2);
+  ASSERT_EQ(mat_store_.capacity(), cap_);
+  double qty_exp = 2 * mat1_->quantity();
+  double space_exp = cap_ - qty_exp;
+  ASSERT_EQ(mat_store_.space(), space_exp);
+  ASSERT_EQ(mat_store_.quantity(), qty_exp);
+
+  Material::Ptr pop1_ = mat_store_.Pop();
+  ASSERT_EQ(pop1_->comp(), test_comp1_);
+  Material::Ptr pop2_ = mat_store_.Pop();
+  ASSERT_EQ(pop2_->comp(), test_comp2_);
+
+}
+
+TEST_F(MaterialBufTest, BulkTest) {
+
+  ASSERT_EQ(bulk_store_.count(), 1);
+  ASSERT_EQ(bulk_store_.capacity(), cap_);
+  double qty_exp = 2 * mat1_->quantity();
+  double space_exp = cap_ - qty_exp;
+  ASSERT_EQ(bulk_store_.space(), space_exp);
+  ASSERT_EQ(bulk_store_.quantity(), qty_exp);
+
+  Material::Ptr pop1_ = mat_store_.Pop();
```

Comment:
Tests pass after fixing this 
```suggestion
  Material::Ptr pop1_ = bulk_store_.Pop();
```